web/src/lib/utils/helpers.tsx

```diff
@@ -239,13 +247,16 @@ export const camelCaseToSentence = (text: string): string => {
   return spacedCase.charAt(0).toUpperCase() + spacedCase.slice(1);
 };
 const sectionsToTasksMap = (roadmap: Roadmap | undefined): Record<SectionType, Task[]> | undefined =>
-  roadmap?.steps.reduce((accumulator, currentValue: Step) => {
-    accumulator[currentValue.section] = [...(accumulator[currentValue.section] || []), ...currentValue.tasks];
+  roadmap?.steps.reduce((accumulator, currentStep: Step) => {
+    const currentStepTasks = roadmap.tasks.filter((task) => task.stepNumber === currentStep.step_number);
+    accumulator[currentStep.section] = [...(accumulator[currentStep.section] || []), ...currentStepTasks];
     return accumulator;
   }, {} as Record<SectionType, Task[]>);
 
 const stepInRoadmap = (roadmap: Roadmap | undefined, taskId: string): Step | undefined =>
-  roadmap?.steps.find((step) => step.tasks.find((task) => task.id === taskId));
+  roadmap?.steps.find(
+    (step) => step.step_number === roadmap?.tasks.find((task) => task.id === taskId)?.stepNumber
```

Comment:
I think it might be nice (more readable, for one) to break out the found task `roadmap?.tasks.find((task) => task.id === taskId)` as its own variable and then do a guard clause in case we don't find it.

In addition to the readability of not having nested finds, it also protects us from a weird situation. I know that `step.step_number` should never be undefined. But imagine if it was somehow. The right side of the `===` has a bunch of `?` in it, which means it could evaluate to undefined. Which means that if `step.step_number` ever was undefined somehow, then undefined === undefined, and it would evaluate as true, and we would return that step despite it not actually being a match. 

Imo, better to guard clause if `roadmap?.tasks.find((task) => task.id === taskId)` is undefined before we do this find.